src/libraries/Common/tests/System/Security/Cryptography/AlgorithmImplementations/AES/AesCipherTests.OneShot.cs

```diff
@@ -0,0 +1,638 @@
+                byte[] decrypted = aes.DecryptEcb(encrypted, padding);
+
+                if (padding == PaddingMode.Zeros)
+                {
+                    Assert.Equal(plaintext, decrypted[..plaintext.Length]);
+                    AssertFilledWith(0, plaintext.AsSpan(plaintext.Length));
+                }
+                else
+                {
+                    Assert.Equal(plaintext, decrypted);
+                }
+
+                decrypted = aes.DecryptEcb(ciphertext, padding);
+                encrypted = aes.EncryptEcb(decrypted, padding);
+
+                if (padding == PaddingMode.ISO10126)
+                {
+                    int blockSizeBytes = aes.BlockSize / 8;
+                    Assert.Equal(ciphertext[..^blockSizeBytes], encrypted[..^blockSizeBytes]);
+                }
+                else
+                {
+                    Assert.Equal(ciphertext, encrypted);
+                }
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(EcbTestCases))]
+        public static void TryDecryptEcb_DestinationTooSmall(byte[] plaintext, byte[] ciphertext, PaddingMode padding)
+        {
+            if (plaintext.Length == 0)
+            {
+                // Can't have a ciphertext length shorter than zero.
+                return;
+            }
+
+            using (Aes aes = AesFactory.Create())
+            {
+                aes.Key = s_aes128OneShotKey;
+
+                Span<byte> destinationBuffer = new byte[plaintext.Length - 1];
+
+                bool result = aes.TryDecryptEcb(ciphertext, destinationBuffer, padding, out int bytesWritten);
+                Assert.False(result, "TryDecryptEcb");
+                Assert.Equal(0, bytesWritten);
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(EcbTestCases))]
+        public static void TryEncryptEcb_DestinationTooSmall(byte[] plaintext, byte[] ciphertext, PaddingMode padding)
+        {
+            if (ciphertext.Length == 0)
+            {
+                // Can't have a too small buffer for zero.
+                return;
+            }
+
+            using (Aes aes = AesFactory.Create())
+            {
+                aes.Key = s_aes128OneShotKey;
+
+                Span<byte> destinationBuffer = new byte[ciphertext.Length - 1];
+
+                bool result = aes.TryEncryptEcb(plaintext, destinationBuffer, padding, out int bytesWritten);
+                Assert.False(result, "TryDecryptEcb");
+                Assert.Equal(0, bytesWritten);
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(EcbTestCases))]
+        public static void TryDecryptEcb_DestinationJustRight(byte[] plaintext, byte[] ciphertext, PaddingMode padding)
+        {
+            using (Aes aes = AesFactory.Create())
+            {
+                aes.Key = s_aes128OneShotKey;
+
+                int expectedPlaintextSize = padding == PaddingMode.Zeros ? ciphertext.Length : plaintext.Length;
```

Comment:
This seems weird.  PaddingMode.Zeros /can/ mean that plaintext.Length == ciphertext.Length, but only PaddingMode.None guarantees it.

Reply:
Oh, nevermind, I get it.  On decrypt PaddingMode.Zeros can't remove the padding, so it's always ciphertext.Length.